core/field_number.js

```diff
@@ -31,22 +31,23 @@ goog.require('Blockly.FieldTextInput');
 
 /**
  * Class for an editable number field.
- * @param {(string|number)=} opt_value The initial content of the field.
- *     The value should cast to a number, and if it does not, '0' will be used.
+ * @param {string|number=} opt_value The initial value of the field. Should cast
+ *    to a number. Defaults to 0.
  * @param {(string|number)=} opt_min Minimum value.
  * @param {(string|number)=} opt_max Maximum value.
  * @param {(string|number)=} opt_precision Precision for value.
- * @param {Function=} opt_validator An optional function that is called
- *     to validate any constraints on what the user entered.  Takes the new
- *     text as an argument and returns either the accepted text, a replacement
- *     text, or null to abort the change.
- * @extends {Blockly.FieldTextInput}
+ * @param {Function=} opt_validator A function that is called to validate
+ *    changes to the field's value. Takes in a number & returns a validated
+ *    number, or null to abort the change.
  * @constructor
  */
 Blockly.FieldNumber = function(opt_value, opt_min, opt_max, opt_precision,
     opt_validator) {
   this.setConstraints(opt_min, opt_max, opt_precision);
-  opt_value = (opt_value && !isNaN(opt_value)) ? String(opt_value) : '0';
+  opt_value = this.doClassValidation_(opt_value);
+  if (opt_value === null) {
+    opt_value = 0;
```

Comment:
What if 0 is outside of the constraints?  Maybe set it to the nearest constraint (min if 0 is below min, max if 0 is above max).

@RoboErikG any preference?

Reply:
Or do we just say eh, creating it with an invalid initial value is a bad idea? I think there's a similar question for all fields--what if the default value for a text field is the empty string but the validator requires it to be nonempty, for instance?  Is there even a reasonable response?

Reply:
I'll wait for @RoboErikG 's response before I change things.